libevm/legacy/legacy_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+package legacy
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/libevm/core/vm"
+)
+
+// stubPrecompileEnvironment implements [vm.PrecompileEnvironment] for testing.
+type stubPrecompileEnvironment struct {
+	vm.PrecompileEnvironment
+	gasToReturn uint64
```

Comment:
(fyi) This is ok for the stub but just commenting so you understand how the real thing works. The production implementation deducts from the total remaining, returning false from `UseGas()` once entirely depleted.

Reply:
Oh god what the hell did I write indeed.
Reworked a bit in https://github.com/ava-labs/libevm/pull/114/commits/b32c8c31931ae04a26e0b094ef08f6cbbdfbee5a